src/frame_buffer.rs

```diff
@@ -165,26 +169,76 @@ impl<'a> FrameBuffer<'a> {
+    fn get_channel(&self, name: &str) -> Option<Channel> {
+        let c_name = CString::new(name.as_bytes()).unwrap();
+        let mut error_out = std::ptr::null();
+        let mut channel = unsafe { mem::uninitialized() };
+        if unsafe {
+               CEXR_FrameBuffer_get_channel(self.handle,
+                                            c_name.as_ptr(),
+                                            &mut channel,
+                                            &mut error_out)
+           } == 0 {
+            Some(channel)
+        } else {
+            None
+        }
+    }
+
+    // TODO: these should probably be part of Header.  They're only here
+    // right now to allow access to both struct's internals, but they won't
     // have to be here for that once `pub(crate)` lands in Rust 1.18.
     //
-    // This shouldn't be used outside of this crate, but due to
+    // These shouldn't be used outside of this crate, but due to
     // https://github.com/rust-lang/rfcs/pull/1422 not being stable
     // yet (should land in Rust 1.18), just hide from public
     // documentation for now.
     // TODO: once Rust 1.18 comes out, change from pub to pub(crate)`.
     #[doc(hidden)]
-    pub fn validate_header_for_output(&self, header: &Header) -> Result<()> {
-        let w = header.data_window();
-        if (w.max.x - w.min.x) as usize != self.dimensions().0 - 1 ||
-           (w.max.y - w.min.y) as usize != self.dimensions().1 - 1 {
-            return Err(Error::Generic(format!("framebuffer size {}x{} does not \
-                match output file dimensions {}x{}",
-                                              self.dimensions().0,
-                                              self.dimensions().1,
-                                              w.max.x - w.min.x,
-                                              w.max.y - w.min.y)));
+    pub fn validate_channels_for_output(&self, header: &Header) -> Result<()> {
+        for chan in header.channels() {
+            let (name, h_channel) = chan?;
+            if let Some(fb_channel) = self.get_channel(name) {
+                FrameBuffer::validate_channel(name, &h_channel, &fb_channel)?;
+            } else {
+                return Err(Error::Generic(format!("FrameBuffer is missing some \
+                    channels that are in Header (needed for output): \
+                    FrameBuffer is missing channel '{}'",
+                                                  name)));
+            }
+        }
+        Ok(())
+    }
+
+    #[doc(hidden)]
+    pub fn validate_channels_for_input(&self, header: &Header) -> Result<()> {
+        for chan in header.channels() {
+            let (name, h_channel) = chan?;
+            if let Some(fb_channel) = self.get_channel(name) {
+                FrameBuffer::validate_channel(name, &h_channel, &fb_channel)?;
+            }
+        }
+        Ok(())
+    }
+
+    fn validate_channel(name: &str, h_chan: &Channel, fb_chan: &Channel) -> Result<()> {
```

Comment:
It doesn't need to be.  I kept it there mainly to keep them spatially proximate, since it's only used in those two methods.  If you feel strongly about moving it out into a stand-alone function I'm happy to do that.  Let me know!